### PR TITLE
[RS][DDCE-1220] Updated data mapping for submission.

### PR DESCRIPTION
**Pull request by RafSobol**
# DDCE-(StoryNumberHere)

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/models/submission/CharityPartnerTransformer.scala [hunk 1]
```diff
@@ -30,52 +30,139 @@ class CharityPartnerTransformer extends JsonTransformer {
...
   def userAnswersToPartner: Reads[JsObject] = {
 
     val partners = for {
       a <- (__ \ 'authorisedOfficials).readNullable(Reads.seq(getOfficials(action, relationAO, partnerTypeIndividual))).map(x => x.fold(JsArray())(JsArray(_)))
       b <- (__ \ 'otherOfficials).readNullable(Reads.seq(getOfficials(action, relationOO, partnerTypeIndividual))).map(x => x.fold(JsArray())(JsArray(_)))
-    } yield a ++ b
+      organisation <- (__ \\ 'organisation).readNullable(getOrganisationNominee(action)).map(x => x.fold(JsArray())(Json.arr(_)))
+      individual <- (__ \\ 'individual).readNullable(getIndividualNominee(action)).map(x => x.fold(JsArray())(Json.arr(_)))
+
+    } yield a ++ b ++ organisation ++ individual
```
Comment (abhijeetardale) [inline app/models/submission/CharityPartnerTransformer.scala:143]: could please rename a and b

File: it/resources/scenario_3_transformed_request.json [hunk 2]
```diff
@@ -1 +1 @@
...
...gdom","nonUKAddress":false}},"type":"1","addPartner":{"effectiveDateOfChange":"T_O_D_A_Y"}}],"charity":{"operationAndFunds":{"incomeReceivedToDate":100,"northernIreland":false,"otherAreaOperation":true,"futureFunds":"donations, fundraising","ukWide":false,"operationAndFundsCommon":{"financialAccounts":true,"accountPeriodEnd":"1111"},"englandAndWales":true,"estimatedGrossIncome":100,"scotland":true,"overseas":true},"charityOrganisation":{"registeredRegulator":true,"regulator":{"otherRegulatorName":"John","otherRegulator":true,"otherRegulatorRegistrationNumber":"Ceana"}},"orgPurpose":{"charitablePurposes":{"other":true,"education":true,"animalWelfare":true,"humanRights":true,"healthOrSavingOfLives":true,"religion":true,"artsCultureOrScience":true,"citizenshipOrCommunityDevelopment":true,"amateurSport":true,"environmentalProtection":true,"reliefOfPoverty":true,"reliefOfYouthAge":true,"armedForcesOfTheCrown":true},"charitableObjectives":{"objectivesA":"Test"},"publicBenefit":{"publicBenefitA":"Test"}},"aboutOrganisation":{"governingApprovedDoc":true,"documentEnclosed":"1","aboutOrgCommon":{"effectiveDate":"2001-11-11"},"governingApprovedChanges":"Test","governingApprovedWords":true}}}}
```
Comment (abhijeetardale) [inline it/resources/scenario_3_transformed_request.json:1]: new line please

**Review by abhijeetardale — CHANGES_REQUESTED** [review]
transformation are mission for WhyNoBankStatementPage(noBankStatement) and OverseasCountry

File: app/models/submission/CharityPartnerTransformer.scala [hunk 3]
```diff
@@ -30,52 +30,139 @@ class CharityPartnerTransformer extends JsonTransformer {
...
-      getName(__ \ 'individualDetails\ 'name, __ \ 'officialsName) and
-      (__ \ 'individualDetails \ 'position).json.copyFrom((__ \ 'officialsPosition ).json.pick) and
-      (__ \ 'individualDetails \ 'dateOfBirth).json.copyFrom((__ \ 'officialsDOB ).json.pick) and
-      (__ \ 'individualDetails \ 'dayPhoneNumber).json.copyFrom((__ \ 'officialsPhoneNumber \ 'daytimePhone).json.pick) and
-      ((__ \ 'individualDetails \ 'mobilePhone).json.copyFrom((__ \ 'officialsPhoneNumber \ 'mobilePhone).json.pick) orElse doNothing) and
-      ((__ \ 'individualDetails \ 'nino).json.copyFrom(nino) orElse doNothing)
+
+    val position = prefix match {
+      case "officials" => (__ \ 'individualDetails \ 'position).json.copyFrom((__ \ s"${prefix}Position").json.pick)
+      case _ => (__ \ 'individualDetails \ 'position).json.put(JsString("01"))
```
[the remark this answers is not in the document]
Comment (RafSobol) [reply app/models/submission/CharityPartnerTransformer.scala:43]: note that it's the same for nominee organisation's authorised person - would we need a position screen for them, too? worth looking into

**Review by abhijeetardale — APPROVED** [review]

**Pull request merged**